avocado/plugins/xunit.py

```diff
@@ -35,7 +36,8 @@ def configure(self, parser):
 
         self.parser = parser
         run_subcommand_parser.output.add_argument(
-            '--xunit', type=str, dest='xunit_output', metavar='FILE',
```

Comment:
I think you can also cleanup the unittests as the FileOrStdoutAction is tested independently, so we might decrease the number of `--xunit - --json -`-like selftest.